# Gene-Expression-Analysis
In the analysis of breast cancer RNASeq data, I started by preparing the TCGA RNASeq data for breast cancer from cbioportal , reading RNASeq, patient, and copy number aberration data. The patient IDs were standardized across these datasets for consistency. Utilizing CNA data, samples were classified based on ERBB2 gene amplification, creating a metadata frame. Next, DESeq2 library was used for the normalization and differential expression analysis, identifying key differentially expressed genes. To identify the 10 Differentially Expressed Genes, log2FoldChange was used. Finally, a PCA plot was generated and pathway enrichment analysis was conducted using the library org.Hs.eg.db which is a Bioconductor annotation data package and enrichKEGG function from the library clusterProfiler, which is used for KEGG Enrichment Analysis of a gene set. 
